frontend/ui/widget/imageviewer.lua

```diff
@@ -155,6 +155,18 @@ function ImageViewer:init()
         self.image = self._scaled_image_func(1) -- native image size, that we need to know
     end
 
+    local image_fit_autorotate = G_reader_settings:readSetting("image_fit_rotate", false)
+    if image_fit_autorotate then
+        -- Automatically rotate the image if it better fits the screen rotated.
+        local aspect_screen = Screen:getWidth() / Screen:getHeight()
+        local aspect_image = self.image:getWidth() / self.image:getHeight()
+        local aspect_misaligned = (aspect_screen <= 1 and aspect_image > 1) or (aspect_screen > 1 and aspect_image <= 1)
+
+        if aspect_misaligned then
```

Comment:
(blank line uneeded)
Again, misaligned feels like not the right word. May be this?:
```
local matching_ar_orientation = ...
if not matching_ar_orientation then
```